content/gsoc/2022.md

```diff
@@ -0,0 +1,600 @@
+
+### Assess requirements for midterm/final evaluation
+- 1st term: Implement events everywhere in the relevant places across Rizin code and event-driven
+	interaction with Cutter.
+- Final term: Add support for the Cutter interface refresh based on the events from Rizin, implement analysis in background.
+
+### Mentors
+- thestr4ng3r
+- Karliss
+
+## Heap viewer (Medium project)
+
+We already have a nice heap (and memory map) parser and visualizer in Rizin (`dm` and `dmh` commands). After debugging becomes a first-class citizen in cutterland it would be awesome to have memory map and heap visualizations.
+
+### Task
+ - Expose Rizin API/commands for Cutter to use for visualization
+ - Design and implement heap navigation and inspection widgets
+ - Provide the integration with current debugging mode in Cutter
+ - Make the implementation work with both local (native) and remote debugging modes
+
+### Skills
+The participant should be comfortable with the C++, and be familiar with Qt framework
+
+### Difficulty
+Medium
+
+### Benefits for the participant
+The participant will gain the understanding on how modern runtimes provide the heap for various
+programs, which will be beneficial for the binary exploitation skills.
+
+### Benefits for the project
+It will greatly improve the debugging and reverse engineering experience for complex programs,
+also provides the way to design the exploitation techniques with the help of Rizin/Cutter.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: Design and implement heap visualization widgets, add Rizin test and fixes
+- Final term: Various bugfixes related to the heap inspection support on various platforms and allocators, tests and documentation.
+
+### Mentors
+- xvilka
+- Megabeets
+
+### Links/Resources
+- [Issue #1041](https://github.com/rizinorg/cutter/issues/1041)
+- [Heap Viewer plugin for IDA Pro](https://github.com/danigargu/heap-viewer)
+- [Heap parsing for MacOS, tmalloc, jmalloc](https://github.com/rizinorg/rizin/issues/208)
+- [Dynamic Allocator Detection](https://github.com/rizinorg/rizin/issues/157)
+- ["heap"-marked Rizin issues](https://github.com/rizinorg/rizin/labels/heap)
+
+## Diffing mode (Medium project)
+
+Binary diffing is one of the most common tasks for the reverse engineer. There are many various
+tools available, but most of them are either detached from the main RE toolbox or poorly integrated.
+Rizin provides basic diffing features out of the box with `rz-diff` tool, but Cutter has no
+interface to represent similar functionality.
+
+### Task
+- Expose basic `rz-diff` features in the Cutter
+- Create the interface to choose two files for diffing
+- Create the way to show the differences in all main widgets:
+	- Hexadecimal view
+	- Disassembly view
+	- Graph view
+	- Pseudocode view
+
+### Skills
+The participant should be comfortable with the C++ language, and be familiar with Qt framework
+
+### Difficulty
+Advanced
```

Comment:
i think this is medium. hex diffing is slightly more difficult than asm/grap/pseudo diffing.
i would be nice to have the results of rz-diff for function similarities on 2 bins into cutter